deployment/helm/node-feature-discovery/values.yaml

```diff
@@ -399,6 +399,8 @@ topologyUpdater:
   kubeletPodResourcesSockPath:
   updateInterval: 60s
   watchNamespace: "*"
+  kubeletStateDir:
```

Comment:
We should actually wire this in `templates/topologyupdater.yaml` so that changing the setting has some effect.

You also want to document this new parameter in `docs/deployment/helm.md` 😉

Reply:
@marquiz I think we should have the default value `/host-var/lib/kubelet` here otherwise it won't get mounted.
Mount it by default without checking the value is problematic if we want to disable the feature when empty string - `""` is provided. WDYT?

Reply:
Hmm, yes, disabling this is a bit problematic. I think there are two obvious options if just using one setting (and not a second one like `disableKubeletStateWatcher`)
1. Set the default here like `kubeletStateDir: /host-var/lib/kubelet`, then setting it to `""`would disable it
2. Set the default to null (`kubeletStateDir: null`) in which case use the default mount, `""`would disable it and ` != ""` would be user-specified mount/dir

I think option 1 above would probably as it's simpler and easier to comprehend. Also, we need to sync the mount dir to the default value of `-kubelet-state-dir` flag anyway so why not put it into the values.yaml file

Reply:
> Hmm, yes, disabling this is a bit problematic. I think there are two obvious options if just using one setting (and not a second one like `disableKubeletStateWatcher`)
> 
> 1. Set the default here like `kubeletStateDir: /host-var/lib/kubelet`, then setting it to `""`would disable it
> 2. Set the default to null (`kubeletStateDir: null`) in which case use the default mount, `""`would disable it and ` != ""` would be user-specified mount/dir
> 
> I think option 1 above would probably as it's simpler and easier to comprehend.
> 
> EDIT: dropped my comment regarding `-kubelet-state-dir` flag which was incorrect

Great so option 1 is exactly what I did :)